lib/middleware/user.js

```diff
@@ -43,8 +55,16 @@ router.use((req, res, next) => {
     })
     .then(profile => {
       req.user.profile = profile;
-      next();
     })
+    .then(() => {
+      if (!req.user.profile.emailConfirmed && req.user._auth.email_verified) {
+        return confirmEmail()
+          .then(() => {
+            req.user.profile.emailConfirmed = true;
+          });
+      }
```

Comment:
Is this triggering an email verification if they're already verified on keycloak but not locally? Are we doing a one off trigger for everyone on their next login?

Reply:
This detects if they are already verified in keycloak and marks their email as verified in our database if that is the case.

It has two purposes:
1) it allows ASRU to continue use the flag in the keycloak admin console to manually verify emails.
2) it prevents all existing ccounts from needing to re-verify when we deploy the feature.